cmd/syncer/main.go

```diff
@@ -212,6 +215,46 @@ func initSyncerComponents(ctx context.Context, clusterFlavor cnstypes.CnsCluster
 				log.Errorf("failed to initialize nodeManager. Error: %+v", err)
 				os.Exit(1)
 			}
+			if configInfo.Cfg.Global.ClusterDistribution == "" {
+				config, err := rest.InClusterConfig()
+				if err != nil {
+					log.Errorf("failed to get InClusterConfig: %v", err)
+				}
+				clientset, err := kubernetes.NewForConfig(config)
+				if err != nil {
+					log.Errorf("failed to create kubernetes client with err: %v", err)
```

Comment:
do we want to exit the container if we are not able to create k8s client to fetch the version?

Reply:
Right, I left this as an open item to discuss on the PR.
I think so we should exit the container. See if the latest commit addresses the concern